src/edfi-paging-test/edfi_paging_test/api/request_client.py

```diff
@@ -0,0 +1,239 @@
+        self.oauth = OAuth2Session(client=client)
+
+    def _build_url_for_resource(self, resource) -> str:
+        return f"/data/v3/ed-fi/{resource}"
+
+    def _build_query_params_for_page(self, page_index, page_size: int) -> str:
+        page_offset = (page_index - 1) * page_size
+        return f"offset={page_offset}&limit={page_size}"
+
+    def _build_query_params_for_total_count(self) -> str:
+        return f"offset={0}&limit={0}&totalCount=true"
+
+    def _authorize(self) -> None:
+        self.oauth.fetch_token(self.api_base_url + OAUTH_TOKEN_URL, auth=self.auth)
+
+    def _check_response(
+        self, response: Response, success_status: HTTPStatus, http_method: str, url: str
+    ) -> None:
+        """
+        Check a response for success
+
+        Parameters
+        ----------
+        response: Response
+            The HTTP response to check
+        success_status: HTTPStatus
+            The HTTP status that indicates success
+        http_method: str
+            A human-readable string describing the http method, used for logging
+        url: str
+            The url of the request, used for logging
+
+        Raises
+        -------
+        RuntimeError
+            If the response indicates failure
+        """
+        if response.status_code != success_status:
+            raise RuntimeError(
+                f"{response.reason} ({response.status_code}): {response.text}"
+            )
+
+    def _get(self, relative_url: str) -> Response:
+        """
+        Send an HTTP GET request.
+
+        Parameters
+        ----------
+        relative_url : str
+            The resource endpoint that you want to request.
+
+        Returns
+        -------
+        dict
+            A parsed response from the server
+
+        Raises
+        -------
+        RuntimeError
+            If the GET operation is unsuccessful
+        """
+        assert isinstance(
+            self.api_base_url, str
+        ), "Property `api_base_url` should be of type `str`."
+
+        if not self.oauth.authorized:
+            self._authorize()
+
+        url = self.api_base_url + relative_url
+        try:
+            response = self.oauth.get(
+                url=url,
+                auth=self.oauth.auth,
+            )
+        except TokenExpiredError:
+            self._authorize()
+            # Re-raise the exception to trigger an automated retry
+            raise
+
+        self._check_response(
+            response=response, success_status=HTTPStatus.OK, http_method="GET", url=url
+        )
+        return response
+
+    @retry(
+        retry_on_exceptions=(
+            IOError,
+            TokenExpiredError,
+            ConnectionError,
+            RequestException,
+            HTTPError,
+            ProtocolError,
+            Timeout,
+            RuntimeError,
+            socket.timeout,
+            socket.error,
+        ),
+        max_calls_total=retry_count,
```

Comment:
This shouldn't be working - a decorator can't access an instance method. I'm very confused by this.